pkg/controller/worker/genericactuator/machine_controller_manager.go

```diff
@@ -48,29 +48,17 @@ func (a *genericActuator) deployMachineControllerManager(ctx context.Context, wo
 	}
 	injectPodAnnotation(mcmValues, "checksum/secret-machine-controller-manager", util.ComputeChecksum(mcmKubeconfigSecret.Data))
 
-	// If the shoot is hibernated then we want to scale down the machine-controller-manager. However, we want to first allow it to delete
-	// all remaining worker nodes. Hence, we cannot set the replicas=0 here (otherwise it would be offline and not able to delete the nodes).
-	if extensionscontroller.IsHibernated(cluster.Shoot) {
-		deployment := &appsv1.Deployment{}
-		if err := a.client.Get(ctx, kutil.Key(workerObj.Namespace, a.mcmName), deployment); err != nil && !apierrors.IsNotFound(err) {
-			return err
-		}
-		if replicas := deployment.Spec.Replicas; replicas != nil {
-			mcmValues["replicas"] = *replicas
-		}
+	replicaCount, err := replicas()
+	if err != nil {
+		return err
 	}
+	mcmValues["replicas"] = replicaCount
 
 	if err := a.mcmSeedChart.Apply(ctx, a.chartApplier, workerObj.Namespace,
 		a.imageVector, a.gardenerClientset.Version(), cluster.Shoot.Spec.Kubernetes.Version, mcmValues); err != nil {
 		return errors.Wrapf(err, "could not apply MCM chart in seed for worker '%s'", util.ObjectName(workerObj))
 	}
 
-	if !controller.IsHibernated(cluster.Shoot) {
-		if err := a.applyMachineControllerManagerShootChart(ctx, workerDelegate, workerObj, cluster); err != nil {
```

Comment:
Why not keeping this here? I guess it is related to the deployment of MCM itself, so why did you move it to the actuator reconciliation function?

Reply:
I moved it because of the `!controller.IsHibernated(cluster.Shoot)` condition. But since you said we didn't need it anymore, I'll move it back.